### PR TITLE
Set METAFLOW_RUNTIME_NAME for executions by AWS Step Functions

### PR DESCRIPTION
**Pull request by savingoyal**

### CODE REVIEW

File: metaflow/plugins/aws/batch/batch.py [hunk 1]
```diff
@@ -159,8 +159,6 @@ def create_job(
             # is NOT set (see get_datastore_root_from_config in datastore/local.py).
         for name, value in env.items():
             job.environment_variable(name, value)
-        for name, value in self.metadata.get_runtime_environment('batch').items():
```
Comment (romain-intel) [inline metaflow/plugins/aws/batch/batch.py:162]: Why are you removing this? The commit message says that you are adding a new environment variable but this seems to be removing the setting of said env var. Is it done somewhere else?

Comment (savingoyal) [reply metaflow/plugins/aws/batch/batch.py:162]: This was overwriting the env values set by step-functions. That's why I yanked it out. Currently, we don't have any `batch` specific runtime environment anyways.

Comment (savingoyal) [reply metaflow/plugins/aws/batch/batch.py:162]: See this - https://github.com/Netflix/metaflow/blob/master/metaflow/plugins/aws/step_functions/step_functions.py#L521

Comment (romain-intel) [reply metaflow/plugins/aws/batch/batch.py:162]: Got it.

**Review by romain-intel — APPROVED** [review]
LGTM

**Pull request merged**